beacon-chain/blockchain/receive_blob.go

```diff
@@ -0,0 +1,22 @@
+package blockchain
+
+import "github.com/prysmaticlabs/prysm/v4/container/slice"
+
+// SendNewBlobEvent sends a message to the BlobNotifier channel that the blob
+// for the blocroot `root` is ready in the database
+func (s *Service) SendNewBlobEvent(root [32]byte, index uint64) {
+	s.blobNotifier.Lock()
+	var ok bool
+	var nc blobNotifierChan
+	nc, ok = s.blobNotifier.chanForRoot[root]
+	if !ok {
+		nc = blobNotifierChan{indices: []uint64{index}, channel: make(chan struct{})}
+		s.blobNotifier.chanForRoot[root] = nc
+	} else {
+		if !slice.IsInUint64(index, nc.indices) {
+			nc.indices = append(nc.indices, index)
```

Comment:
this shouldn't be in an else, you'll miss the first index where the map entry is created

Reply:
the index was added in that creation as well wasn't it?

Reply:
oh right the problem is if it's created in the reader